Sources/Common/CustomerIOInstance.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+public protocol CustomerIOInstance: AutoMockable {
+    var siteId: String? { get }
+
+    /// Get the current configuration options set for the SDK.
+    var config: SdkConfig? { get }
+
+    func identify(
+        identifier: String,
+        body: [String: Any]
+    )
+
+    // sourcery:Name=identifyEncodable
+    // sourcery:DuplicateMethod=identify
+    func identify<RequestBody: Encodable>(
+        identifier: String,
+        // sourcery:Type=AnyEncodable
+        // sourcery:TypeCast="AnyEncodable(body)"
+        body: RequestBody
+    )
+
+    var registeredDeviceToken: String? { get }
+
+    func clearIdentify()
+
+    func track(
+        name: String,
+        data: [String: Any]
+    )
+
+    // sourcery:Name=trackEncodable
+    // sourcery:DuplicateMethod=track
+    func track<RequestBody: Encodable>(
+        name: String,
+        // sourcery:Type=AnyEncodable
+        // sourcery:TypeCast="AnyEncodable(data)"
+        data: RequestBody?
+    )
+
+    func screen(
+        name: String,
+        data: [String: Any]
+    )
+
+    // sourcery:Name=screenEncodable
+    // sourcery:DuplicateMethod=screen
+    func screen<RequestBody: Encodable>(
+        name: String,
+        // sourcery:Type=AnyEncodable
+        // sourcery:TypeCast="AnyEncodable(data)"
+        data: RequestBody?
+    )
+
+    var profileAttributes: [String: Any] { get set }
+    var deviceAttributes: [String: Any] { get set }
+
+    func registerDeviceToken(_ deviceToken: String)
+
+    func deleteDeviceToken()
+
+    func trackMetric(
+        deliveryID: String,
+        event: Metric,
+        deviceToken: String
```

Comment:
🤔 I had an idea? 

The reason for moving this interface into the Common module is for the CDP module to be able to add new public functions this interface (and I think CustomerIO class?). 

These functions highlighted here I think are functions that are directly tied to the Track module, right? 

Would it be worth having the `CustomerIOInstance` in the Common module be almost empty and then have the Track module add all of these functions via Track module?

Reply:
That's a very good point, this interface does have indeed things that are `tracking` module only and they would indeed need to be moved into implementation by `tracking` modules. 

I'll think a bit more about this to make sure, we are not adding redundancy while also adding the `data pipeline` specific extension. But i do see alot of value in keeping it bare minimum and letting track then add more specific implementation.